nhais-adaptor/adaptor/incoming/tests/test_operation_definition_adaptor.py

```diff
@@ -0,0 +1,94 @@
+import unittest
+from testfixtures import compare
+import adaptor.incoming.operation_definition_adaptor as incoming_adaptor
+from edifact.incoming.models.interchange import Interchange, InterchangeHeader
+from edifact.incoming.models.message import Messages, MessageSegment, MessageSegmentBeginningDetails, \
+    MessageSegmentRegistrationDetails
+import adaptor.fhir_helpers.fhir_creators as creators
+
+
+class TestOperationDefinitionAdaptor(unittest.TestCase):
+    """
+    Test the conversion of an incoming edifact interchange to a fhir operation definition response
+    """
+
+    def test_format_date_time(self):
+        """
+        Tests the function that formats the edifact date time stamp to a fhir format
+        """
+        formatted_date = incoming_adaptor.format_date_time("190501:0902")
+        self.assertEqual(formatted_date, "2019-05-01 09:02")
+
+    def test_create_operation_definition(self):
+        """
+        Tests the function to create a fhir operation definition from an incoming edifact interchange
+        """
+        with self.subTest("adapt an interchange with one message to a single fhir operation definition"):
+            sender_parameter = creators.create_parameter_with_binding(name="senderCypher", value="XX11",

```

Comment:
There's a lot of setting up of the interchange in these test cases. I think it's probably worth extracting this to a test helper method to make the tests read more clearly and reduce duplication.

Reply:
Yeah will do. I'm finding most of the work and time is being spent in setting up the fixture or expected data for these tests😂. I'm still trying to see if there is a more concise way of doing this.